internal/service/managedobjectstorage/policy.go

```diff
@@ -2,183 +2,345 @@ package managedobjectstorage
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "Policy name.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"service_uuid": schema.StringAttribute{
+				Description: "Managed Object Storage service UUID.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"system": schema.BoolAttribute{
+				Description: "Defines whether the policy was set up by the system.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"updated_at": schema.StringAttribute{
+				Description: "Update time.",
+				Computed:    true,
+			},
 		},
-		Schema: schemaPolicy(),
 	}
 }
 
-func schemaPolicy() map[string]*schema.Schema {
-	return map[string]*schema.Schema{
-		"arn": {
-			Description: "Policy ARN.",
-			Computed:    true,
-			Type:        schema.TypeString,
-		},
-		"attachment_count": {
-			Description: "Attachment count.",
-			Computed:    true,
-			Type:        schema.TypeInt,
-		},
-		"created_at": {
-			Description: "Creation time.",
-			Computed:    true,
-			Type:        schema.TypeString,
-		},
-		"default_version_id": {
-			Description: "Default version id.",
-			Computed:    true,
-			Type:        schema.TypeString,
-		},
-		"description": {
-			Description: "Description of the policy.",
-			Optional:    true,
-			ForceNew:    true,
-			Type:        schema.TypeString,
-		},
-		"document": {
-			Description: "Policy document, URL-encoded compliant with RFC 3986.",
-			Required:    true,
-			ForceNew:    true,
-			Type:        schema.TypeString,
-		},
-		"name": {
-			Description: "Policy name.",
-			Required:    true,
-			ForceNew:    true,
-			Type:        schema.TypeString,
-		},
-		"service_uuid": {
-			Description: "Managed Object Storage service UUID.",
-			Required:    true,
-			ForceNew:    true,
-			Type:        schema.TypeString,
-		},
-		"system": {
-			Description: "Defines whether the policy was set up by the system.",
-			Computed:    true,
-			Type:        schema.TypeBool,
-		},
-		"updated_at": {
-			Description: "Update time.",
-			Computed:    true,
-			Type:        schema.TypeString,
-		},
+func (r *managedObjectStoragePolicyResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = policySchema(1)
+}
+
+func setPolicyValues(_ context.Context, data *policyModel, policy *upcloud.ManagedObjectStoragePolicy) diag.Diagnostics {
+	var respDiagnostics diag.Diagnostics
+
+	data.ARN = types.StringValue(policy.ARN)
+	data.AttachmentCount = types.Int64Value(int64(policy.AttachmentCount))
+	data.CreatedAt = types.StringValue(policy.CreatedAt.String())
```

Comment:
Check that this (and `UpdatedAt`) uses correct string formatting

Reply:
`String()` has been used also before for these fields, but should we change it to `Format(time.RFC3339)` here or in the next major release? 🤔

Reply:
I would vote for the major release on this one.